src/ui/inventory.rs

```diff
@@ -33,28 +32,235 @@ pub fn build_menu(api: &mut api::API, machine_index: i32) {
+      wrefresh(win);
+      refresh();
+      //let requested_machine = getch();
+      //TODO: something. Drop drink I guess.
+    
+      let slot_count = slots.len();
+      let mut selected_slot: i32 = 0;
 
-  let mut slot_count = 1; // Start printing machines on the 3rd row of the Window.
-  for slot in inventory.iter() {
-    mvwprintw(
-      win,
-      2 + slot_count,
-      2,
-      format!("{}. {} ({})", slot_count, slot.name, slot.price).as_str(),
-    );
-    slot_count += 1;
+      for n in 0..slot_count {
+          if n as i32 == selected_slot {
+              wattron(win, A_REVERSE());
+          }
+          if slots[n].empty {
+            wattron(win, COLOR_PAIR(1));
+          }
+          if !slots[n].active {
+            wattron(win, A_DIM());
+          }
+          mvwprintw(
+              win, 3 + n as i32, 2,
+              format!("{} ({} credits)", slots[n].name, slots[n].price).as_str(),
+          );
+          wattroff(win, A_DIM());
+          wattroff(win, COLOR_PAIR(1));
+          wattroff(win, A_REVERSE());
+      }
+
+      refresh();
+      wrefresh(win);
+    
+      let mut key = getch();
+      loop {
+        match key {
+            KEY_UP => {
+              if selected_slot > 0 {
+                selected_slot -= 1;
+              }
+            },
+            KEY_DOWN => {
+              if selected_slot < slot_count as i32 - 1 {
+                selected_slot += 1;
+              }
+            },
+            KEY_RIGHT => { 
+              //inventory::build_menu(&mut api, selected_machine);
+              if !slots[selected_slot as usize].empty && slots[selected_slot as usize].active {
+                match api.drop(machine_name.clone(), selected_slot as u8 + 1) { // The API returns a zero-indexed array of slots, but Mizu wants it to be 1-indexed 
+                    Ok(()) => {
+                      vend();
+                      // Refresh credits in case we bought anything.
+                      credits = api::API::get_credits(api);
+                      wmove(win, height-2, width-20);
+                      wclrtoeol(win);
+                      mvwprintw(win, height-2, width-20, format!("Credits: {}", credits.unwrap()).as_str());
+                    },
+                    _ => deny()
+                }
+              }
+              else {
+                deny();
+              }
+            },
+            KEY_LEFT => {
+              ui_common::destroy_win(win);
+              return;
+            },
+            _ => {
+              refresh();
+            }
+        }
+        
+        for n in 0..slot_count {
+          if n as i32 == selected_slot {
+              wattron(win, A_REVERSE());
+          }
+          if slots[n].empty {
+            wattron(win, COLOR_PAIR(1));
+          }
+          if !slots[n].active {
+            wattron(win, A_DIM());
+          }
+          mvwprintw(
+              win, 3 + n as i32, 2,
```

Comment:
Magic numbers?

Reply:
Magic numbers. This is just what ncurses code usually looks like.